src/test/kotlin/org/elm/ide/inspections/inference/TypeDeclarationInspectionTest.kt

```diff
@@ -7,7 +10,7 @@ class TypeDeclarationInspectionTest : ElmInspectionsTestBase(ElmTypeDeclarationI
 
     fun `test bad mutual self-recursion in type alias`() = checkByText("""
 <error descr="Infinite recursion">type alias A = B</error>
-<error descr="Infinite recursion">type alias B = A</error>
+type alias B = A
```

Comment:
The compiler only reports the first error, and not reporting both allows us to cache more inference results.